django/contrib/contenttypes/fields.py

```diff
@@ -243,7 +241,8 @@ def __get__(self, instance, cls=None):
                 rel_obj = ct.get_object_for_this_type(pk=pk_val)
             except ObjectDoesNotExist:
                 pass
-        setattr(instance, self.cache_attr, rel_obj)
+
+        self.set_cached_value(instance=instance, value=rel_obj)
```

Comment:
Fine by me. I think we could probably omit the keywords in a lot of places and include them only where it's potentially confusing, but probably that's already what you're doing.